gapic-generator-cloud/templates/cloud/client_test.erb

```diff
@@ -269,91 +219,35 @@ describe Google::Cloud::Speech::V1::SpeechClient do
-            client.long_running_recognize config, audio
-          end
-
-          # Verify the GaxError wrapped the custom error that was raised.
-          assert_match custom_error.message, err.message
-        end
-      end
-    end
-  end
-
-  describe "streaming_recognize" do
-    let :custom_error do
-      CustomTestErrorV1.new "Custom test error for Google::Cloud::Speech::V1::SpeechClient#streaming_recognize."
-    end
-
-    it "invokes streaming_recognize without error" do
-      # Create request parameters
-      request = {}
-
-      # Create expected grpc response
-      expected_response = {}
-      expected_response = Google::Gax.to_proto expected_response,
-                                               Google::Cloud::Speech::V1::StreamingRecognizeResponse
-
-      # Mock Grpc layer
-      mock_method = proc do |requests|
-        request = requests.first
-        OpenStruct.new execute: [expected_response]
-      end
-      mock_stub = MockGrpcClientStubV1.new :streaming_recognize, mock_method
+<% end %>
+      mock_stub = MockGrpcClientStub<%= namespaces.last.upcase_first %>.new :<%= method_name %>, mock_method
 
       # Mock auth layer
-      mock_credentials = MockSpeechCredentialsV1.new "streaming_recognize"
+      mock_credentials = Mock<%= service.name %>Credentials<%= namespaces.last.upcase_first %>.new "<%= method_name %>"
 
-      Google::Cloud::Speech::V1::Speech::Stub.stub :new, mock_stub do
-        Google::Cloud::Speech::V1::Credentials.stub :default, mock_credentials do
-          client = Google::Cloud::Speech.new version: :v1
-
-          # Call method
-          response = client.streaming_recognize [request]
-
-          # Verify the response
-          assert_equal 1, response.count
-          assert_equal expected_response, response.first
-        end
-      end
-    end
-
-    it "invokes streaming_recognize with error" do
-      # Create request parameters
-      request = {}
-
-      # Mock Grpc layer
-      mock_method = proc { raise custom_error }
-      mock_stub = MockGrpcClientStubV1.new :streaming_recognize, mock_method
-
-      # Mock auth layer
-      mock_credentials = MockSpeechCredentialsV1.new "streaming_recognize"
-
-      Google::Cloud::Speech::V1::Speech::Stub.stub :new, mock_stub do
-        Google::Cloud::Speech::V1::Credentials.stub :default, mock_credentials do
-          client = Google::Cloud::Speech.new version: :v1
+      <%= ruby_namespace service.address %>::Stub.stub :new, mock_stub do
+        <%= ruby_namespace namespaces %>::Credentials.stub :default, mock_credentials do
+          client = <%= ruby_namespace service.address[0..-3] %>.new version: :<%= namespaces.last.downcase %>
 
           # Call method
           err = assert_raises Google::Gax::GaxError do
-            client.streaming_recognize [request]
+<% if method_streaming_bidi? method %>
+            client.<%= method_name %> [request]
+<% else %>
+            client.<%= method_name %> config, audio
```

Comment:
won't generalize

Reply:
I can't think of how else to generalize `client.recognize` except with `client.<%= method_name %>`. Can you explain a bit?

Reply:
That part is okay. It's the method arguments (`config, audio`) that will have to change.